File: ReadRecyclerAdapter.java
package com.example.fitpeak;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import java.util.ArrayList;

public class ReadRecyclerAdapter extends RecyclerView.Adapter<ReadRecyclerAdapter.ReadViewHolder>
{
    private Context context;
    private ArrayList<fData> fitList;

    public ReadRecyclerAdapter(Context context, ArrayList<fData> fitList)
    {
        this.context = context;
        this.fitList = fitList;
    }

    public class ReadViewHolder extends RecyclerView.ViewHolder
    {
        TextView personName, personActivity, activityData;

        public ReadViewHolder(View itemView)
        {
            super(itemView);
            personName = itemView.findViewById(R.id.txtName);
            personActivity = itemView.findViewById(R.id.txtPersonActivity);
            activityData = itemView.findViewById(R.id.txtQuantity);
        }
    }

    @Override
    public ReadViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
    {
        View view = LayoutInflater.from(context).inflate(R.layout.single_data_item, parent, false);
        return new ReadViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ReadViewHolder holder, int position)
    {
        holder.personName.setText(fitList.get(position).getPersonName());
        holder.personActivity.setText(fitList.get(position).getPersonActivity());
        holder.activityData.setText(String.valueOf(fitList.get(position).getActivityData()));
    }

    @Override
    public int getItemCount()
    {
        return fitList.size();
    }
}

File: ReadRecyclerAdapter.kt
package com.pritampattanayak.tutorial

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class ReadRecyclerAdapter(val context: Context, private val bookList:ArrayList<Book>)
    :RecyclerView.Adapter<ReadRecyclerAdapter.ReadViewHolder>() {

    class ReadViewHolder(view: View):RecyclerView.ViewHolder(view){
        val bookName:TextView=view.findViewById(R.id.txtBookName)
        val bookAuthor:TextView=view.findViewById(R.id.txtBookAuthor)
        val bookPrice:TextView=view.findViewById(R.id.txtBookPrice)
        val bookRating:TextView=view.findViewById(R.id.txtBookRating)

    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReadViewHolder {
       val view=LayoutInflater.from(parent.context).inflate(R.layout.single_book_item,parent,false)
        return ReadViewHolder(view)
    }

    override fun onBindViewHolder(holder: ReadViewHolder, position: Int) {
        holder.bookName.text=bookList[position].bookName
        holder.bookAuthor.text=bookList[position].bookAuthor
        holder.bookPrice.text=bookList[position].bookPrice.toString()
        holder.bookRating.text=bookList[position].bookRating
    }

    override fun getItemCount(): Int {
        return bookList.size
    }
}